controller/volume_controller.go

```diff
@@ -1578,10 +1578,16 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 				if vc.isVolumeUpgrading(v) {
 					continue
 				}
-				if r.Spec.FailedAt == "" {
-					r.Spec.FailedAt = vc.nowHandler()
+				notReady, err := vc.ds.IsNodeNotReady(r.Spec.NodeID)
+				if err != nil {
+					return err
+				}
```

Comment:
If we plan to reply on the monitoring and feedback of the engine process, does this controller still need `canIMLaunchReplica, err := vc.canInstanceManagerLaunchReplica(r)`? If NO, then can the controller directly check if the replica is unknown or not?

Reply:
Do you mean directly checking the replica is in unknown state and don't need to rely on `canInstanceManagerLaunchReplica`?

When the replica is in unknown state, the node might be `existing but not ready` or `deleted`, we still use `canInstanceManagerLaunchReplica` to distinguish the two cases and check if we can stop the replica.

Reply:
If the node is unavailable, the engine process should mark the replica as ERR then report to the engine monitor thread in the engine controller. In other words, the volume controller does not need to take care of everything. This simplifies the whole control. WDYT?

Reply:
Agreed. I removed the check `faileAt` and `stopped` from here. The two fields will be updated based on the engine monitor's results.